release-content/0.15/release-notes/15800_Add_mesh_picking_backend_and_MeshRayCast_system_parameter.md

```diff
@@ -1,4 +1,54 @@
 <!-- Add mesh picking backend and `MeshRayCast` system parameter -->
 <!-- https://github.com/bevyengine/bevy/pull/15800 -->
 
-<!-- TODO -->
+TODO: Laser beam image
+
+Being able to click on objects to select them is a vital and seemingly simple task in any game.
+Since 2020, doing this in Bevy has largely meant pulling in `@aevyrie`'s beloved ecosystem crate, [`bevy_mod_picking`] and it's simple raycasting companion [`bevy_mod_raycast`].
+
+Over the years, this crate has been refined and battle-tested, both by [Foresight Spatial Labs] (a CAD-creating, Bevy-using company co-founded by Aevyrie) and
+the broader open source community of game developers that have used it for everything from first-person-shooters to point-and-click adventures.
+Bevy is thrilled to have had the chance to work with the team behind [`bevy_mod_picking`] and have adopted the project wholesale into Bevy itself.
+Integrating a large project is a ton of work, and we're incredibly grateful to the contributors who have made `bevy_picking` a stable, first-class feature of the engine.
+
+The new `bevy_picking` crate follows the existing modular architecture closely:
+
+1. Inputs are gathered from mouse, touch and pen devices. Each pointing device (humans are equipped with 10 by default) gets a screen-space [`PointerLocation`].
+2. Each modular [backend] performs the domain-specific work (like raycasting) of figuring out how these pointer locations map to [`PointerHits`] on objects that they're watching.
+3. The hit information from each backend is combined and sorted to produce a coherent [`HoverMap`], which lists which entities each pointer is hovering over.
+4. High level events (both ordinary events and observers!) are emitted for each hovered entity, capturing complex behavior such as clicking, dragging or releasing various objects.
+
+In Bevy 0.15, we're shipping with UI, sprite and mesh backends. Each of these comes with its own caveats for now:
+
+- UI: both the legacy [`Interaction`] and new [`PickingInteraction`] components exist [for now](https://github.com/bevyengine/bevy/issues/15550), with subtle behavioral differences.
+- Sprites: picking always uses the full rectangle, and [alpha transparency is not taken into account](https://github.com/bevyengine/bevy/issues/14929).
+- Mesh: this is a naive raycast against the full mesh, and can be painfully slow. You should be using simplified meshes and an acceleration data structure like a BVH to speed this up if you are using this functionality for games. As a result, this functionality is disabled by default.
```

Comment:
I probably wouldn't say "painfully slow", just "relatively slow" or even "slow for large scenes with lots of geometry". There's still AABB culling and a lot of parallelism, and even without a BVH, it was taking just 0.34% of frame time for the Caldera hotel in my quick (not very thorough) [test](https://github.com/bevyengine/bevy/pull/15800#issuecomment-2408999324). Of course that's just a single ray per frame though.

But yeah, it's also important to make it clear that it's not intended as a replacement for proper ray casting based on colliders, and it should mainly be used for debugging, editor use cases, or simple applications that don't have too much geometry.

Reply:
```suggestion
- Mesh: this is a naive raycast against the full mesh, and can be relatively slow. You should be using simplified meshes and an acceleration data structure like a BVH to speed this up if you are using this functionality for performance-sensitive games. As a result, this functionality is disabled by default.
```